base/acme/src/main/java/org/dogtagpki/acme/server/ACMEEngine.java

```diff
@@ -162,6 +167,47 @@ public void setIssuer(ACMEIssuer issuer) {
         this.issuer = issuer;
     }
 
+    /**
+     * Whether the whole ACME service is enabled or not.
+     */
+    public boolean isEnabled() {
+        return this.enabled;
+    }
+
+    private void setEnabled(boolean b) {
+        this.enabled = b;
+    }
+
+    public void loadEngineConfig(Properties cfg) throws Exception {
+        // the default class just sends the default config values
+        Class<? extends ACMEEngineConfigSource> configSourceClass
+            = ACMEEngineConfigDefaultSource.class;
+
+        String className = cfg.getProperty("engine.class");
+        if (className != null && !className.isEmpty()) {
+            configSourceClass =
+                (Class<ACMEEngineConfigSource>) Class.forName(className);
+        }
+        engineConfigSource = configSourceClass.newInstance();
+
+        // We pass to the ConfigSource only the callbacks needed to set
+        // the configuration (Consumer<T>).  This abstraction ensures the
+        // ConfigSource has no direct access to the ACMEEngine instance.
+        engineConfigSource.init(
+            cfg,
+            new Consumer<Boolean>() {
+                @Override public void accept(Boolean b) {
+                    setEnabled(b);
+                    logger.info(
+                        "ACME service is "
+                        + (b ? "enabled" : "DISABLED")
```

Comment:
nit: same casing?

Reply:
I wanted to make DISABLED stand out a bit more in the log :)